web/react/components/admin_console/user_item.jsx

```diff
@@ -61,9 +77,31 @@ export default class UserItem extends React.Component {
 
     handleMakeAdmin(e) {
         e.preventDefault();
+        const me = UserStore.getCurrentUser();
+        if (this.props.user.id === me.id) {
+            this.handleDemote(this.props.user, 'admin');
+        } else {
+            const data = {
+                user_id: this.props.user.id,
+                new_roles: 'admin'
+            };
+
+            Client.updateRoles(data,
+                () => {
+                    this.props.refreshProfiles();
+                },
+                (err) => {
+                    this.setState({serverError: err.message});
+                }
+            );
+        }
+    }
+
+    handleMakeSystemAdmin(e) {
```

Comment:
@coreyhulen Why handleMakeSystemAdmin should have the handleDemote? handleMakeSystemAdmin doesn't demote role, because SystemAdmin is most powerful role.